starlette/testclient.py

```diff
@@ -452,42 +473,55 @@ def websocket_connect(
         return session
 
     def __enter__(self) -> "TestClient":
-        loop = asyncio.get_event_loop()
-        self.send_queue = asyncio.Queue()  # type: asyncio.Queue
-        self.receive_queue = asyncio.Queue()  # type: asyncio.Queue
-        self.task = loop.create_task(self.lifespan())
-        loop.run_until_complete(self.wait_startup())
+        self.exit_stack = contextlib.ExitStack()
+        self.portal = self.exit_stack.enter_context(
+            anyio.start_blocking_portal(**self.async_backend)
+        )
+        self.stream_send = StapledObjectStream(
+            *anyio.create_memory_object_stream(math.inf)
+        )
+        self.stream_receive = StapledObjectStream(
+            *anyio.create_memory_object_stream(math.inf)
+        )
+        try:
+            self.task = self.portal.spawn_task(self.lifespan)
+            self.portal.call(self.wait_startup)
```

Comment:
Just a thought: could `portal.start_task()` be used here?

Reply:
I think the intention here is that we want to wait for 'wait_startup' to finish before moving on.  And if it fails we want to not continue.

Reply:
Couldn't `self.lifespan()` signal such readiness on its own? Do we need a separate method for that?

Reply:
Well, we want to not leave `TestClient.__enter__` until `wait_startup()` has finished.  I could signal readiness some other way, but I don't understand what we'd gain doing that.

Reply:
Looking at the actual code, it doesn't seem very easy at least.

Reply:
Yeah, never mind.